### PR TITLE
Refactor the Build

### PR DESCRIPTION
**Pull request by mhamilton723**

### CODE REVIEW

**Comment by elibarzilay** [comment]
This is a mix of several different things: there are at least

* Random style changes,
* Directory structure change,
* Import cleanups and dirtyings,

They should really be in separate PRs.

**Comment by elibarzilay** [comment]
I much prefer `!= None` over `.isDefined`.  (Since the latter is a horrible name.)

**Comment by elibarzilay** [comment]
Unclear whether it was a typo, but getting paths like `src/core/src/main/scala/com.microsoft.ml.spark.core/test/benchmarks/` looks pretty horrifying.

**Comment by elibarzilay** [comment]
There are several places with

    import com.microsoft.ml.spark.core.env.FileUtilities._
    import com.microsoft.ml.spark.core.env.FileUtilities.File

**Comment by elibarzilay** [comment]
`src/core/build.sbt` has no newline at the EOF.

**Comment by elibarzilay** [comment]
I don't get the purpose of `NamespaceInjections`.

File: src/core/src/main/scala/com.microsoft.ml.spark.core/test/base/SparkSessionFactory.scala [hunk 1]
```diff
@@ -37,17 +38,39 @@ object SparkSessionFactory {
...
+      activeSessions +=1
+      if (session.isEmpty) {
+        val conf = new SparkConf()
+          .setAppName("MMLSpark")
+          .setMaster("local[*]")
+          .set("spark.logConf", "true")
+          .set("spark.sql.warehouse.dir", SparkSessionFactory.localWarehousePath)
+        val sess = SparkSession.builder()
+          .config(conf)
+          .getOrCreate()
```
Comment (mhamilton723) [inline src/core/src/main/scala/com.microsoft.ml.spark.core/test/base/SparkSessionFactory.scala]: Make this work with newSession

Comment (mhamilton723) [reply src/core/src/main/scala/com.microsoft.ml.spark.core/test/base/SparkSessionFactory.scala]: Done

**Pull request closed without merging**